src/hotspot/share/services/threadService.cpp

```diff
@@ -1477,7 +1478,14 @@ oop ThreadSnapshotFactory::get_thread_snapshot(jobject jthread, TRAPS) {
 
     carrier_thread = Handle(THREAD, java_lang_VirtualThread::carrier_thread(thread_h()));
     if (carrier_thread != nullptr) {
+      // Note: this java_thread may not be protected by the ThreadsListHandle above,
+      // but as we have disabled transitions, if we are mounted on it, then it can
+      // not terminate and so is safe to handshake with.
       java_thread = java_lang_Thread::thread(carrier_thread());
+    } else {
+      // We may have previously found a carrier but since unmounted, so
+      // clear that previous reference.
```

Comment:
nit typo: s/but since unmounted/but it since unmounted.

Reply:
I believe the grammar is correct. It could have said `have since unmounted`, but the `have` is not essential. However as I am expanding the later comment I may as well adjust this one.